feature_engine/creation/distance_features.py

```diff
@@ -0,0 +1,241 @@
+            [coordinate[2] for coordinate in columns],
+            [coordinate[3] for coordinate in columns],
+        )
+
+    def _check_output_columns_names(
+        self,
+        column_name: Optional[List[Union[str, None]]],
+        coordinate_columns: List[List[Union[str, int]]],
+    ) -> Optional[List[Union[str, None]]]:
+        if column_name is None:
+            return [f"distance_{c[0]}_{c[1]}_{c[2]}_{c[3]}" for c in coordinate_columns]
+        if len(column_name) != len(coordinate_columns):
+            raise ValueError(
+                "Not enough output column names provided.\n "
+                f"Expected {len(coordinate_columns)} column names, "
+                f"but got {len(column_name)}."
+            )
+        return column_name
+
+    def fit(self, X: pd.DataFrame):
+        # there is no fit for this transformer
+        super().fit(X)
+        return self
+
+    def transform(self, X: pd.DataFrame):
+        """
+        Compute the distance on heart using the Haversine formula.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shame = [n_samples, n_features]
+            The data to be transformed.
+
+        Returns
+        -------
+        X_new: Pandas dataframe.
+            The original dataframe plus the distances between the given coordinates.
+        """
+        X = self._check_transform_input_and_state(X)
+        X = self._check_lat_lon_columns_are_in_df(X)
+        X = self._check_correctness_of_coordinates(X)
+
+        self._compute_distance(X)
+
+        if self.drop_original:
+            X.drop(
+                columns=[
+                    *self.a_latitudes,
+                    *self.a_longitudes,
+                    *self.b_latitudes,
+                    *self.b_longitudes,
+                ],
+                inplace=True,
+            )
+
+        return X
+
+    def _check_lat_lon_columns_are_in_df(self, X) -> pd.DataFrame:
+        coordinate_columns = [
+            *self.a_latitudes,
+            *self.a_longitudes,
+            *self.b_latitudes,
+            *self.b_longitudes,
+        ]
+        if set(coordinate_columns).issubset(set(X.columns)) is False:
+            raise ValueError(
+                f"The columns {set(coordinate_columns).issubset(set(X.columns))} "
+                f"were not found in the dataframe."
+            )
+        return X
+
+    def _check_correctness_of_coordinates(self, X: pd.DataFrame) -> pd.DataFrame:
+        # recall that the latitude is a number between -90 and +90,
+        # while longitudes is between -180 and +180.
+        irregular_latitudes = (
+            (X[[*self.a_latitudes, *self.b_latitudes]].abs() > 90).sum().sum()
+        )
+        irregular_longitudes = (
+            (X[[*self.a_longitudes, *self.b_longitudes]].abs() > 180).sum().sum()
+        )
+
+        if irregular_latitudes > 0:
+            raise ValueError("The dataframe contains irregular latitudes")
```

Comment:
User may not be familiar with what irregular latitudes are. We need more informative errors. If the data has irregular lats and longs, this error will come first, and we will not let the user know that they also have irregular lons. We could unify the error and return both error messages together